src/super_gradients/common/sg_loggers/dagshub_sg_logger.py

```diff
@@ -0,0 +1,222 @@
+        """
+
+        self.paths = {
+            "dvc_directory": Path("artifacts"),
+            "models": Path("models"),
+            "artifacts": Path("artifacts"),
+        }
+
+        token = dagshub.auth.get_token()
+        os.environ["MLFLOW_TRACKING_USERNAME"] = token
+        os.environ["MLFLOW_TRACKING_PASSWORD"] = token
+
+        # Check mlflow environment variable is set:
+        if not self.repo_name or not self.repo_owner:
+            self.repo_name, self.repo_owner = self.splitter(input("Please insert your repository owner_name/repo_name:"))
+
+        if not self.remote or "dagshub" not in os.getenv("MLFLOW_TRACKING_URI"):
+            dagshub.init(repo_name=self.repo_name, repo_owner=self.repo_owner)
+            self.remote = os.getenv("MLFLOW_TRACKING_URI")
+
+        self.repo = Repo(
+            owner=self.remote.split(os.sep)[-2],
+            name=self.remote.split(os.sep)[-1].replace(".mlflow", ""),
+            branch=os.getenv("BRANCH", "main"),
+        )
+        self.dvc_folder = self.repo.directory(str(self.paths["dvc_directory"]))
+
+        mlflow.set_tracking_uri(self.remote)
+        mlflow.set_experiment(self.experiment_name)
+        self.run = mlflow.start_run(nested=True)
+        return self.run
+
+    @multi_process_safe
+    def _dvc_add(self, local_path="", remote_path=""):
+        if not os.path.isfile(local_path):
+            FileExistsError(f"Invalid file path: {local_path}")
+        self.dvc_folder.add(file=local_path, path=remote_path)
+
+    @multi_process_safe
+    def _dvc_commit(self, commit=""):
+        self.dvc_folder.commit(commit, versioning="dvc", force=True)
+
+    @multi_process_safe
+    def add_config(self, tag: str, config: dict):
+        super(DagsHubSGLogger, self).add_config(tag=tag, config=config)
+        param_keys = config.keys()
+        for pk in param_keys:
+            for k, v in config[pk].items():
+                try:
+                    mlflow.log_params({k: v})
+                except:
+                    logger.warning(f"Skip to log {k}: {v}")
+
+    @multi_process_safe
+    def add_scalar(self, tag: str, scalar_value: float, global_step: int = 0):
+        super(DagsHubSGLogger, self).add_scalar(tag=tag, scalar_value=scalar_value, global_step=global_step)
+        mlflow.log_metric(key=tag, value=scalar_value, step=global_step)
+
+    @multi_process_safe
+    def add_scalars(self, tag_scalar_dict: dict, global_step: int = 0):
+        super(DagsHubSGLogger, self).add_scalars(tag_scalar_dict=tag_scalar_dict, global_step=global_step)
+        mlflow.log_metrics(metrics=tag_scalar_dict, step=global_step)
+
+    @multi_process_safe
+    def close(self):
+        super().close()
+        try:
+            if not self.log_mlflow_only:
+                self._dvc_commit(commit=f"Adding all artifacts from run {mlflow.active_run().info.run_id}")
```

Comment:
Note that although this approach will be faster during training, it might be risky when training on a Spot instance in AWS (or equivalent in other cloud providers).
In such cases, the process has only limited time to stop, and uploading the files only on process closing may cause some files to be lost. 

Consider maybe calling commit every x epochs...

Reply:
Great point, let me explain the logic here:
1. MLflow is the "log everything" tool here, so after every epoch, when we add checkpoints, I log everything (including the model) to MLflow
2. DVC is the reproducibility tool, so I only want to commit things I'm likely to need to reproduce, therefore I'm logging the final and best results to it. The user can always log their artifacts manually if they want, but this makes sure they don't make their git history too overwhelming.

If the user is using spot instances and it disconnects in the middle, they will have all the previous checkpoints logged to MLflow.

Reply:
Got it. thanks